packages/interpretations/src/components/interpretations/InterpretationDialog.js

```diff
@@ -17,10 +19,16 @@ const styles = {
 class InterpretationDialog extends Component {
     constructor(props) {
         super(props);
-        this.state = { value: props.interpretation ? props.interpretation.text : "", showEditor: true };
+        this.state = {
+            value: props.interpretation ? props.interpretation.text : "",
+            showEditor: true,
+            sharingDialogIsOpen: false,
+        };
         this.save = this.hideEditorAndThen(this._save.bind(this));
         this.cancel = this.hideEditorAndThen(this._cancel.bind(this));
         this.onChange = this.onChange.bind(this);
+        this.openSharingDialog = this.openSharingDialog.bind(this);
```

Comment:
I also dislike all this boilerplate, but using arrow functions as props is not advisable, as those new functions created on the fly will make components re-render needlessly (since the prop has changed). Alternatives:

1) https://www.npmjs.com/package/react-autobind or similar
2) Use class properties: https://github.com/facebook/react/issues/9851

Reply:
Yes, alternative 2 is what i meant. 

Doing "() => someFunction" as props is not advised.